assets/js/push.js

```diff
@@ -122,6 +148,46 @@ jQuery( window ).on( 'load', () => {
 	function distributorMenuEntered() {
 		distributorMenuItem.focus();
 		document.body.classList.toggle( 'distributor-show' );
+
+		if ( distributorPushWrapper.classList.contains( 'loaded' ) ) {
+			return;
+		}
+
+		distributorPushWrapper.classList.add( 'loaded' );
+
+		const data = {
+			action: 'dt_load_connections',
+			loadConnectionsNonce: dt.loadConnectionsNonce,
+			postId: dt.postId
+		};
+
+		const xhr = dt.usexhr ? { withCredentials: true } : false;
+
+		jQuery.ajax( {
+			url: dt.ajaxurl,
+			xhrFields: xhr,
+			method: 'post',
+			data: data
+		} ).done( ( response ) => {
+			setTimeout( () => {
+				if ( ! response.success || ! response.data ) {
+					doError();
+					return;
+				}
+
+				dtConnections = response.data;
+
+				distributorPushWrapper.innerHTML = processTemplate( 'dt-show-connections' )( {
+					connections: dtConnections,
+				} );
+
+				setVariables();
+			}, 500 );
+		} ).error( () => {
+			setTimeout( () => {
+				doError();
```

Comment:
maybe remove the loaded class from `distributorPushWrapper` on failure so load will try again